tcf_website/static/common/sorting.js

```diff
@@ -20,6 +20,38 @@ function cmpByProp(prop, asc) {
         if (prop === "date") { // convert date string to int
             valA = dateToInt(valA);
             valB = dateToInt(valB);
+        } else if (prop === "recency") {
+            try {
+                const sessionA = valA.match(/^[a-zA-Z]+/)[0];
+                valA = valA.match(/-?[0-9]\d*(\.\d+)?/)[0];
+                // last taught session taken into account for sorting
+                if (sessionA === "January") {
+                    valA = parseFloat(valA + "0");
+                } else if (sessionA === "Spring") {
+                    valA = parseFloat(valA + "1");
+                } else if (sessionA === "Summer") {
+                    valA = parseFloat(valA + "2");
+                } else {

```

Comment:
I'm not sure if this is best practice or excessively obvious documentation, so feel free to ignore this if you disagree, but it might be helpful to make it explicit that the `else` here _should_ only ever be the Fall semester. Maybe even do another `if` and change the `else` to throw an error if we get unexpected data.

Reply:
I added another `if` to handle the Fall semester, and also changed the `else` to sort unexpected data to the end.